lib/core/src/Cardano/Pool/DB/Sqlite/TH.hs

```diff
@@ -85,34 +83,28 @@ PoolOwner sql=pool_owner
 
 -- Mapping of registration certificate to pool
 PoolRegistration sql=pool_registration
-    poolRegistrationPoolId             W.PoolId  sql=pool_id
-    poolRegistrationSlot               W.SlotId  sql=slot
-    poolRegistrationMarginNumerator    Word64    sql=margin_numerator
-    poolRegistrationMarginDenominator  Word64    sql=margin_denominator
-    poolRegistrationCost               Word64    sql=cost
+    poolRegistrationPoolId             W.PoolId                      sql=pool_id
+    poolRegistrationSlot               W.SlotId                      sql=slot
+    poolRegistrationMarginNumerator    Word64                        sql=margin_numerator
+    poolRegistrationMarginDenominator  Word64                        sql=margin_denominator
+    poolRegistrationCost               Word64                        sql=cost
+    poolRegistrationPledge             Word64                        sql=pledge
+    poolRegistrationMetadataUrl        Text Maybe                    sql=metadata_url
+    poolRegistrationMetadataHash       W.StakePoolMetadataHash Maybe sql=metadata_hash
 
     Primary poolRegistrationPoolId
     deriving Show Generic
 
--- Temporary queue where metadata to fetch are stored.
-PoolMetadataQueue sql=pool_metadata_queue
-    poolMetadataQueuePoolId            W.PoolId           sql=pool_id
-    poolMetadataQueueUrl               Text               sql=metadata_url
-    poolMetadataQueueHash              ByteString         sql=metadata_hash
-
-    Primary poolMetadataQueuePoolId
-    deriving Show Generic
-
 -- Cached metadata after they've been fetched from a remote server.
 PoolMetadata sql=pool_metadata
-    poolMetadataPoolId                 W.PoolId           sql=pool_id
-    poolMetadataName                   Text               sql=name
-    poolMetadataTicker                 W.StakePoolTicker  sql=ticker
-    poolMetadataDescription            Text Maybe         sql=description
-    poolMetadataHomepage               Text               sql=homepage
-    poolMetadataPledge                 Word64             sql=pledge
-    poolMetadataOwner                  ByteString         sql=owner
-
-    Primary poolMetadataPoolId
+    poolMetadataHash                   W.StakePoolMetadataHash sql=metadata_hash
+    poolMetadataPoolId                 W.PoolId                sql=pool_id
+    poolMetadataName                   Text                    sql=name
+    poolMetadataTicker                 W.StakePoolTicker       sql=ticker
+    poolMetadataDescription            Text Maybe              sql=description
+    poolMetadataHomepage               Text                    sql=homepage
+
+    Primary poolMetadataHash
+    Foreign PoolRegistration fk_registration_metadata_hash poolMetadataPoolId ! ON DELETE CASCADE
```

Comment:
```
Foreign PoolRegistration fk_registration_metadata_hash poolMetadataPoolId ! ON DELETE CASCADE
```
Does this delete the metadata when the registration certificate with the same pool id is rolled deleted/rolled back?

Reply:
Yes.

Reply:
Which might not be desirable actually.